libs/common-services/src/main/java/uk/gov/di/ipv/core/library/service/YamlConfigService.java

```diff
@@ -56,6 +73,22 @@ public YamlConfigService(File parametersFile, File secretsFile) {
         }
     }
 
+    private JsonNode getParamsWithCorrectClientConfig(File parametersFile, File apiClientConfigFile)
+            throws IOException {
+        try {
+            var params = YAML_OBJECT_MAPPER.readTree(parametersFile).get(CORE);
+
+            if (apiClientConfigFile != null) {
+                var apiClientConfig = YAML_OBJECT_MAPPER.readTree(apiClientConfigFile).get(CLIENTS);
+                ((ObjectNode) params).replace(CLIENTS, apiClientConfig);
+            }
```

Comment:
Left it like this for simplicity as I know we're only overriding the client config for the api tests but maybe this could override explicit config defined in the yaml (similar to how dev-deploy works with its user configs) instead of just the client? Couldn't find a handy package to do this in java though so might need to make a util from scratch.